python/trezorlib/tests/device_tests/test_msg_recoverydevice_shamir.py

```diff
@@ -221,3 +221,140 @@ def test_abort(self):
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Enter first share
+        assert ret == proto.ButtonRequest(code=proto.ButtonRequestType.MnemonicInput)
+        self.client.transport.write(proto.ButtonAck())
+        for word in mnemonics[0].split(" "):
+            time.sleep(1)
+            self.client.debug.input(word)
+        ret = self.client.transport.read()
+
+        # Homescreen
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_yes()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        for i in range(5):
+            assert ret == proto.ButtonRequest(
+                code=proto.ButtonRequestType.MnemonicInput
+            )
+            self.client.transport.write(proto.ButtonAck())
+            time.sleep(1)
+            if i == 0:
+                # enter first word wrong (different from previous share)
+                self.client.debug.input(mnemonics[0].split(" ")[-1])
+            elif i == 1:
+                # enter second word wrong (different from previous share)
+                self.client.debug.input(mnemonics[0].split(" ")[0])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[-1])
+            elif i == 2:
+                # enter third word wrong (different from previous share)
+                self.client.debug.input(mnemonics[0].split(" ")[0])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[1])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[-1])
+            elif i == 3:
+                # enter fourth word wrong (same as previous share)
+                self.client.debug.input(mnemonics[0].split(" ")[0])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[1])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[2])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[3])
+            elif i == 4:
+                # enter an invalid share
+                self.client.debug.input(mnemonics[0].split(" ")[0])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[1])
+                time.sleep(1)
+                self.client.debug.input(mnemonics[0].split(" ")[2])
+                time.sleep(1)
+                for _ in range(17):
+                    self.client.debug.input("academic")
+                    time.sleep(1)
+            ret = self.client.transport.read()
+
+            # Confirm warning message
+            assert isinstance(ret, proto.ButtonRequest)
+            assert ret == proto.ButtonRequest(proto.ButtonRequestType.Warning)
+            self.client.debug.press_yes()
+            ret = self.client.call_raw(proto.ButtonAck())
+
+        # TODO: do we want to do this? it will add 40s+ to testing time in case of 3of6
```

Comment:
This just finishes the recovery? I don't think we need that, since we have tests for that above.